components/places/src/api/matcher.rs

```diff
@@ -140,28 +140,34 @@ pub fn accept_result(conn: &PlacesDb, search_string: &str, url: &Url) -> Result<
 }
 
 pub fn split_after_prefix(href: &str) -> (&str, &str) {
-    match memchr::memchr(b':', href.as_bytes()) {
+    // Only search up to 64 bytes (matches desktop behavior)
+    let haystack = &href.as_bytes()[..href.len().min(64)];
+    match memchr::memchr(b':', haystack) {
         None => ("", href),
         Some(index) => {
             let hb = href.as_bytes();
             let mut end = index + 1;
             if hb.len() >= end + 2 && hb[end] == b'/' && hb[end + 1] == b'/' {
                 end += 2;
             }
-            (&href[0..end], &href[end..])
+            href.split_at(end)
         }
     }
 }
 
 pub fn split_after_host_and_port(href: &str) -> (&str, &str) {
```

Comment:
For clarity, split_after_host_and_port is the only one here that's important, although I noticed `split_after_prefix`'s behvior differed from desktop.